tuf/pyca_crypto_keys.py

```diff
@@ -0,0 +1,1017 @@
+      message = 'The private key is encrypted with an unsupported algorithm.'
+      raise tuf.CryptoError(message)
+   
+    # Generate an RSSA-PSS signature.  Raise 'tuf.CryptoError' for the expected
+    # pyca/cryptography exceptions.
+    rsa_signer.update(data)
+    signature = rsa_signer.finalize()
+  
+  else:
+    raise TypeError('The required private key is unset.')
+
+  return signature, method
+
+
+
+
+
+def verify_rsa_signature(signature, signature_method, public_key, data):
+  """
+  <Purpose>
+    Determine whether the corresponding private key of 'public_key' produced
+    'signature'.  verify_signature() will use the public key, signature method,
+    and 'data' to complete the verification.
+    
+    >>> public, private = generate_rsa_public_and_private(2048)
+    >>> data = b'The quick brown fox jumps over the lazy dog'
+    >>> signature, method = create_rsa_signature(private, data)
+    >>> verify_rsa_signature(signature, method, public, data)
+    True
+    >>> verify_rsa_signature(signature, method, public, b'bad_data')
+    False
+
+  <Arguments>
+    signature:
+      An RSASSA PSS signature, as a string.  This is the signature returned
+      by create_rsa_signature(). 
+
+    signature_method:
+      A string that indicates the signature algorithm used to generate
+      'signature'.  'RSASSA-PSS' is currently supported.
+
+    public_key:
+      The RSA public key, a string in PEM format.
+
+    data:
+      Data used by tuf.keys.create_signature() to generate
+      'signature'.  'data' (a string) is needed here to verify 'signature'.
+
+  <Exceptions>
+    tuf.UnknownMethodError.  Raised if the signing method used by
+    'signature' is not one supported by tuf.keys.create_signature().
+    
+    tuf.FormatError. Raised if 'signature', 'signature_method', or 'public_key'
+    are improperly formatted.
+
+  <Side Effects>
+    pyca/cryptography's RSAPublicKey.verifier() called to do the actual
+    verification.
+
+   <Returns>
+    Boolean.  True if the signature is valid, False otherwise.
+  """
+  
+  # Does 'public_key' have the correct format?
+  # This check will ensure 'public_key' conforms to 'tuf.formats.PEMRSA_SCHEMA'.
+  # Raise 'tuf.FormatError' if the check fails.
+  tuf.formats.PEMRSA_SCHEMA.check_match(public_key)
+
+  # Does 'signature_method' have the correct format?
+  tuf.formats.NAME_SCHEMA.check_match(signature_method)
+
+  # Does 'signature' have the correct format?
+  tuf.formats.PYCRYPTOSIGNATURE_SCHEMA.check_match(signature)
+  
+  # What about 'data'?
+  tuf.formats.DATA_SCHEMA.check_match(data)
+
+  # Verify whether the private key of 'public_key' produced 'signature'.
+  # Before returning the 'valid_signature' Boolean result, ensure 'RSASSA-PSS'
+  # was used as the signing method.
+  valid_signature = False
+
+  # Verify the signature with pyca/cryptography if the signature method is
+  # valid, otherwise raise 'tuf.UnknownMethodError'.
+  if signature_method == 'RSASSA-PSS':
```

Comment:
Instead of the long `if` clause, why not get unsupported signature methods out of the way upfront, just as is done with the `SCHEMA` checks? Decreases indent, untangles pre-flight checks and actual business logic:
```
if signature_method != "RSASSA-PSS":
  raise tuf.UnknownMethodError(signature_method)
# Implicit `else` follows....
```